frontend/pkg/frontend/frontend.go

```diff
@@ -185,24 +188,43 @@ func (f *Frontend) ArmResourceList(writer http.ResponseWriter, request *http.Req
 	if resourceGroupName != "" {
 		prefixString += "/resourceGroups/" + resourceGroupName
 	}
+	if resourceName != "" {
+		// This is a nested resource request. Build a resource ID for
+		// the parent cluster. We use this below to get the cluster's
+		// ResourceDocument from Cosmos DB.
+		prefixString += "/providers/" + api.ProviderNamespace
+		prefixString += "/" + api.ClusterResourceTypeName + "/" + resourceName
+	}
 	prefix, err := arm.ParseResourceID(prefixString)
 	if err != nil {
 		f.logger.Error(err.Error())
 		arm.WriteInternalServerError(writer)
 		return
 	}
 
-	documentList, continuationToken, err := f.dbClient.ListResourceDocs(ctx, prefix, &api.ClusterResourceType, pageSizeHint, continuationToken)
-	if err != nil {
-		f.logger.Error(err.Error())
-		arm.WriteInternalServerError(writer)
-		return
-	}
+	dbIterator := f.dbClient.ListResourceDocs(ctx, prefix, pageSizeHint, continuationToken)
 
 	// Build a map of cluster documents by Cluster Service cluster ID.
 	documentMap := make(map[string]*database.ResourceDocument)
-	for _, doc := range documentList {
-		documentMap[doc.InternalID.ID()] = doc
+	for item := range dbIterator.Items(ctx) {
+		var doc database.ResourceDocument
+
+		err = json.Unmarshal(item, &doc)
+		if err != nil {
+			f.logger.Error(err.Error())
+			arm.WriteInternalServerError(writer)
+			return
+		}
+
+		if strings.HasSuffix(strings.ToLower(doc.Key.ResourceType.Type), resourceTypeName) {
```

Comment:
Can this be done with a `WHERE` clause at the DB layer?

Reply:
Technically yes, but see my in-memory cache answer below.

I'm going to leave this as is for this PR but I added a "FIXME" comment here to circle back to it once I've gotten rid of the generic database interface I described below.